### ISSUE TITLE
Too Long Pre-Training Makes Overfit the Training Dataset?

### ISSUE THREAD

**Issue by GloryyrolG** [NVlabs/VAEBM#3]
Hi @authors, @xavierxiao, @arash-vahdat ,

May 400-epoch pre-training of NVAE be too long? Since I found, the model overfits the training dataset and has an extremely high NELBO on the val dataset, although the number is consistent with the original NVAE practice. I was using one 24GB GPU. Thanks.

![image](https://user-images.githubusercontent.com/33202229/147172557-dd7cabb1-9fc7-4cf6-9a55-f5673834d7e4.png)

![image](https://user-images.githubusercontent.com/33202229/147172508-5c4293ab-b254-4c57-a903-73fafac6178b.png)

**Comment by arash-vahdat** [comment]
The issue you are seeing with validation NELBO is most likely not an overfitting case. This behavior is due to the instability of NVAE at the evaluation time. It is most likely caused by the running mean and variance of the batch norm layers not converging. My guess is that if you continue this run with a small learning rate or if you evaluate the model in the training mode instead of the eval mode, you will be able to successfully evaluate this model.

Regarding the original question of if 400 epochs are too many: if you change the batch size, I would recommend changing the number of epochs such that the model is trained for a similar number of total iterations. For example, if the total batch size is halved, you can usually reduce the number of epochs by a factor of 2.
